modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/cards/get_data_attributes.js

```diff
@@ -15,7 +15,7 @@
 export default function getDataAttributes(data) {
 	return data
 		? Object.entries(data).reduce((acc, [key, value]) => {
-				acc[`data-${key}`] = value;
+				acc[`data-${key.toLowerCase()}`] = value;
```

Comment:
This is the warning I'm talking about in case it's not clear

![](https://user-images.githubusercontent.com/5572/98825711-19d22080-2435-11eb-8754-a9ebca844825.png)

Reply:
And by the way, I have a feeling that this error


![](https://user-images.githubusercontent.com/5572/98827998-cf05d800-2437-11eb-912e-8c45c8a2d951.png)


 that we've been seeing for a while could be fixed if [`DropdownMenu.js`](https://github.com/liferay/liferay-portal/blob/b50d81b60bf1437cc1418227d8d96eaefefce314/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/DropdownMenu.js) would use the [`getDataAttributes`](https://github.com/liferay/liferay-portal/blob/b50d81b60bf1437cc1418227d8d96eaefefce314/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/cards/get_data_attributes.js) function instead of it's own - but that doesn't seem to be part of the scope of this PR.
If you agree though, I can go ahead and create a ticket and send a fix for that.

Reply:
> If you agree though, I can go ahead and create a ticket and send a fix for that.

:+1:

Reply:
As we talked offline I'd not go with this approach since we're renaming stuff and that could cause confusion in developers. For example:

In Java they add to the items some data like `action` or `deleteDepotEntryURL`.

That that is mean to be used in the `propsTransformer`

```
submitForm(document.hrefFm, itemData.deleteDepotEntryURL);
```
If we transform those data attributes to lowerCase then we will receive `itemData.deletedepotentryurl` there, which could be confusion for the developer.

Reply:
What I don't know is how is this working right now 😅

Reply:
@carloslancha I agree that renaming stuff is probably a dangerous move.

if we're accessing an element's attributes (I don't know if that's the case though) there's no difference, i.e.

```js
var el = document.createElement('div');
el.setAttribute('data-myStuff', 'foo');
console.log(el.getAttribute('data-myStuff')); // "foo" 
console.log(el.getAttribute('data-mystuff')); // "foo" 
```

That said, and as talked on slack,  if we changed the case of the attributes from Java, we'd still be renaming those attributes right?

Reply:
Ok, so after having analysed this and discussed it with @carloslancha we decided we're not going to change anything in the `getDataAttributes` function because we shouldn't, at the risk of breaking things.

We're also thinking that we are using the `data` attributes incorrectly so we could fix that in the future.

Reply:
To expand what @julien said

The thing is we were adding some `data` to the items in order to use that that in the `defaultEventHandlers` (like the action name and needed parameters like `deleteDepotEntryURL`), but they're not meant to be rendered as data-attributes, and we don't needed now that we have the `propsTransformer` mechanism.

For now I'll leave it as it is, it's just a warning, but I'd suggest to create a new pattern for this. For example, we can add all this info to an object called `additionalData`, that we will use in the `propsTransformer` but that we will not pass through in the js component (`VerticalCard.js` in this case).

In this way we can continue using `data` for passing `data-attributes` when needed and `additionalData` only for those we will need to use in a `propsTransformer`.

Makes sense?

Reply:
cc @jbalsas

Reply:
Not sure where to answer... this is quite a long thread and I haven't read it through.

https://html.spec.whatwg.org/#custom-data-attribute

> A custom data attribute is an attribute in no namespace whose name starts with the string "data-", has at least one character after the hyphen, is XML-compatible, and contains no ASCII upper alphas.

Which is what @julien was exposing in https://github.com/liferay-frontend/liferay-portal/pull/495#discussion_r522088968

Reply:
> Not sure where to answer... 

Yes you're right the "threads" in this PR are really hard to see so from now on @carloslancha and I decided we shouldn't make the conversations as resolved because it makes it hard to see what's going on.

Thanks for the answer, I should have checked the spec and posted that too.
I personally think it would be "nicer" if we didn't have those react warning because seeing that output in the devtools is not a great experience.

@carloslancha do you agree if we change those attributes to lowercase as I was doing initially [here](https://github.com/liferay-frontend/liferay-portal/blob/0a1799a69a67b4381c2c0009aa8dbe50407a8a52/modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/cards/get_data_attributes.js#L18)?

Reply:
As we talked offline, I'd leave it as it is, is a warning about something that is not completely right and that will be solved when this gets migrated to use `propsTransformer` mechanism.

Reply:
@carloslancha yes agreed. I [reverted that change](https://github.com/liferay-frontend/liferay-portal/pull/495/commits/9ce43829036624b42a05986dbb270d7cecd86629) after our conversation.

Reply:
> What I don't know is how is this working right now

It works because **most** browsers do `toLowerCase` automatically.